scalableminds/webknossos
# Add option to lock explorative annotations

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Create an explorative annotation of your favourite dataset
- Go to the explorative annotations view. There should be a button in the actions column to lock the annotation. Click it. It should add the "locked" tag to the annotation.
- View the annotation by clicking the respective action. In the annotation view, you should not be able to annotate the annotation. A "banner" in the top right corner should inform you about the fact that the annotation is locked by you. A tooltip should give information by whom this annotation is locked. 
- Moreover, there should be a menu item to unlock the annotation in the menu header bar. Once clicked, the annotation is unlocked and reloaded.
- In the explorative annotations view, it should not be possible to archive a locked annotation.

- An annotation cannot be locked and unlocked by a user with whom the annotation is shared.

### TODOs:
- [x] Lock all edit possibilities in the Tracing view
- [x] Write migration
- [x] Fix the 
- [x] bug, that after unlocking an annotation in the annotation view, the locked tag was still present in the explorative annotations list.
- [x] Do not allow to archive locked mappings
- [x] In the frontend: Get rid of `allowUpdateAndIsNotLocked` as a locked annotation will always be set to `allowUpdate: false`

### Issues:
- fixes [#5087](https://github.com/scalableminds/webknossos/issues/5087)

------
(Please delete unneeded items, merge only when none are left open)
  - likely yes
